### PR TITLE
Change ISMRGarbageInfo to adopt new form of SMREntry

### PR DESCRIPTION
**Pull request by lixinchengdu**
## Overview

Change ISMRGarbageInfo to adopt new form of SMREntry (modified in https://github.com/CorfuDB/CorfuDB/pull/2102)

### CODE REVIEW

File: runtime/src/main/java/org/corfudb/protocols/logprotocol/SMRRecordGarbageInfo.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+ */
+@ToString(callSuper = true)
+@NoArgsConstructor
+public class SMRRecordGarbageInfo {
+
+    /**
+     * Global address that detected the associated SMREntry as garbage.
+     */
+    @Getter
+    private long detectorAddress;
```
Comment (WenbinZhu) [inline runtime/src/main/java/org/corfudb/protocols/logprotocol/SMRRecordGarbageInfo.java]: Can you change this to `markerAddress` as we discussed before?

File: runtime/src/main/java/org/corfudb/protocols/logprotocol/SMRGarbageInfo.java [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    }
+
+    /**
+     * Get the serialized size of garbage-identified SMREntries that has detector address
+     * up to addressUpTo.
+     *
+     * @param addressUpTo detector address up to which garbage sizes are counted
+     * @return size in Byte.
+     */
+    public int getGarbageSizeUpTo(long addressUpTo) {
```
Comment (WenbinZhu) [inline runtime/src/main/java/org/corfudb/protocols/logprotocol/SMRGarbageInfo.java]: Why `addressUpTo` is not used?

For your reference, following codes are what I used to use:
```java
public int getGarbageSizeUpTo(long addressUpTo) {
        return streamIdToGarbageMap.values().stream()
                .map(multiSmr -> multiSmr.getGarbageSizeUpTo(addressUpTo))
                .reduce(0, (a, b) -> a + b);
}
```

File: runtime/src/main/java/org/corfudb/protocols/logprotocol/SMRRecordGarbageInfo.java (hunk 1 above)
Comment (WenbinZhu) [inline runtime/src/main/java/org/corfudb/protocols/logprotocol/SMRRecordGarbageInfo.java]: Can you change this to `markerAddress` as we discussed before?

File: runtime/src/main/java/org/corfudb/protocols/logprotocol/SMRGarbageInfo.java [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+public class SMRGarbageInfo extends LogEntry {
+
+    /**
+     * A map to maintain information about garbage-identified SMREntry inside MultiObjectSMREntry.
+     * The key of the map is the stream id.
+     */
+    @Getter
+    private final Map<UUID, Map<Integer, SMRRecordGarbageInfo>> streamIdToGarbageMap = new ConcurrentHashMap<>();
+
+    Map<Integer, SMRRecordGarbageInfo> EMPTY_GARBAGE_MAP = new ConcurrentHashMap<>();
```
Comment (WenbinZhu) [inline runtime/src/main/java/org/corfudb/protocols/logprotocol/SMRGarbageInfo.java]: Should this be static?

File: runtime/src/main/java/org/corfudb/protocols/logprotocol/SMRRecordGarbageInfo.java (hunk 1 above)
Comment (lixinchengdu) [reply runtime/src/main/java/org/corfudb/protocols/logprotocol/SMRRecordGarbageInfo.java]: done

File: runtime/src/main/java/org/corfudb/protocols/logprotocol/SMRGarbageInfo.java (hunk 2 above)
Comment (lixinchengdu) [reply runtime/src/main/java/org/corfudb/protocols/logprotocol/SMRGarbageInfo.java]: done

File: runtime/src/main/java/org/corfudb/protocols/logprotocol/SMRRecordGarbageInfo.java (hunk 1 above)
Comment (lixinchengdu) [reply runtime/src/main/java/org/corfudb/protocols/logprotocol/SMRRecordGarbageInfo.java]: done

File: runtime/src/main/java/org/corfudb/protocols/logprotocol/SMRGarbageInfo.java (hunk 3 above)
Comment (lixinchengdu) [reply runtime/src/main/java/org/corfudb/protocols/logprotocol/SMRGarbageInfo.java]: done

File: runtime/src/main/java/org/corfudb/protocols/logprotocol/SMRGarbageInfo.java [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+     */
+    public int getGarbageSize() {
+        return getGarbageSizeUpTo(Long.MAX_VALUE);
+    }
+
+    /**
+     * Get the serialized size of garbage-identified SMREntries that has marker address
+     * up to addressUpTo.
+     *
+     * @param addressUpTo detector address up to which garbage sizes are counted
```
Comment (WenbinZhu) [inline runtime/src/main/java/org/corfudb/protocols/logprotocol/SMRGarbageInfo.java]: detector address -> marker address

Comment (lixinchengdu) [reply runtime/src/main/java/org/corfudb/protocols/logprotocol/SMRGarbageInfo.java]: done

**Pull request merged**